### PR TITLE
Cesarvr/porting mixed to jsc

### PR DESCRIPTION
**Pull request by cesarvr**
This PR add support for the Mixed type for the Javascript Core.

### CODE REVIEW

File: src/common/type_deduction.hpp [hunk 1]
```diff
@@ -15,21 +15,113 @@
...
+    }
+
+    types::Type from(DataType data_type) {
+        int realm_type = static_cast<int>(data_type);
+        return static_cast<types::Type>(realm_type);
+    }
+
+    template <typename T, typename Ctx, typename Val>
+    types::Type typeof(Ctx context, Val& value) {
+        using JS = js::Value<T>;
```
Comment (kneth) [inline src/common/type_deduction.hpp:80]: We often use `using Value = js::Value<T>` in our code.

**Review by kneth — APPROVED** [review]
Minor comments.

File: src/js_mixed.hpp [hunk 2]
```diff
@@ -16,14 +16,13 @@
...
-
+#include <common/type_deduction.hpp>
+#include <common/types.hpp>
 #include <map>
 #include <realm/mixed.hpp>
 #include <type_traits>
 
-#include <common/type_deduction.hpp>
-#include <common/types.hpp>
+#pragma once
```
Comment (fronck) [inline src/js_mixed.hpp:25]: Guard should probably be moved up (see @kneth 's comment)

**Review by fronck — APPROVED** [review]
Just one comment

**Pull request merged**